caom2-search-server/src/main/java/ca/nrc/cadc/caom2/ADQLGenerator.java

```diff
@@ -199,7 +199,15 @@ public String toSQL(final IntervalSearch s, final String col,
             throw new UnsupportedOperationException(
                     "IntervalSearch to SQL (fuzzy match)");
         } else if (CAOM2_ENERGY_UTYPE.equals(s.getName())) {
-            sql = toIntersectSQL(s, getColumnName(CAOM2_ENERGY_UTYPE));
+            // CADC-9369, CADC-1052: call this version of toIntersectSQL so it's a plain
+            // upper & lower bound comparison rather than INTERSECTS, which is returning
+            // invalid results
+            // parameters below are IntervalSearch, upper bound, lower bound
+            sql = toIntervalSQL(s, getColumnName("Plane.energy.bounds.upper"),
+                getColumnName("Plane.energy.bounds.lower"));
+            // TODO: reinstate this when CADC-9369 changes are reverted, after
+            // CADC-9367 is completed
+            //            sql = toIntersectSQL(s, getColumnName(CAOM2_ENERGY_UTYPE));
```

Comment:
Does this not also affect ObsCore?  I see only CAOM-2 related changes.

Reply:
ObsCore already uses toIntervalSQL for energy (around line 213, ADQLGenerator.java):

} else if (OBSCORE_ENERGY_UTYPE.equals(s.getName())) {
            sql = toIntervalSQL(s,
                                getColumnName(OBSCORE_ENERGY_UTYPE + ".LoLimit"),
                                getColumnName(OBSCORE_ENERGY_UTYPE + ".HiLimit"));
                              
... so there was nothing to be changed.